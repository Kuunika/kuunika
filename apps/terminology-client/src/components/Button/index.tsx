import React from 'react';
import { withStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';
import { green, blue, red, brown } from '@material-ui/core/colors';
import styled from 'styled-components';
import { theme } from '../../config/theme';

const themes = {
  theme: {
    color: 'white',
    backgroundColor: theme.color,
    onHoverColor: theme.lightDarkColor
  },
  primary: {
    color: 'white',
    backgroundColor: blue[500],
    onHoverColor: blue[700]
  },
  success: {
    color: 'white',
    backgroundColor: green[500],
    onHoverColor: green[700]
  },
  danger: {
    color: 'white',
    backgroundColor: red[500],
    onHoverColor: red[700]
  },
  warning: {
    color: 'white',
    backgroundColor: brown[500],
    onHoverColor: brown[700]
  },
  default: {
    color: theme.lightColor,
    backgroundColor: 'whitesmoke',
    onHoverColor: '#ebebeb'
  }
};

function Btn(props: Props) {
  const theme = props.theme ? themes[props.theme] : themes.theme;

  const CustomBtn = withStyles(() => ({
    root: {
      color: theme.color,
      textTransform: 'capitalize',
      backgroundColor: theme.backgroundColor,
      borderRadius: '50px',
      padding: '8px 20px',
      '&:hover': {
        backgroundColor: theme.onHoverColor
      }
    }
  }))(Button);

  return (
    <CustomBtn
      style={props.style}
      variant={
        props.theme && props.theme === 'default' ? 'outlined' : 'contained'
      }
      onClick={props.onClick ? props.onClick : () => {}}
      {...props}
    >
      {props.icon ? <IconWrapper>{props.icon}</IconWrapper> : <></>}
      {props.children}
    </CustomBtn>
  );
}

type Props = {
  children: any;
  icon?: any;
  onClick?: any;
  theme?: 'theme' | 'primary' | 'success' | 'danger' | 'warning' | 'default';
  style?: any;
};
export default Btn;

const IconWrapper = styled.div`
  display: inline-block;
  margin-right: 10px;
`;
