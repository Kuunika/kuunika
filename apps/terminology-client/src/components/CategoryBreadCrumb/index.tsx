import React from 'react';
import styled from 'styled-components';
import { theme } from '../../config/theme';

function CategoryBreadCrumb({ data }: Props) {
  return (
    <Wrapper>
      <b>Category: </b>
      {data.map((text, index) => {
        text = text.replace('-', ' ');
        return (
          <span key={text}>
            <Location>{text}</Location>
            {index < data.length - 1 && <Separator />}
          </span>
        );
      })}
    </Wrapper>
  );
}

export default CategoryBreadCrumb;

interface Props {
  data: Array<string>;
}

const Wrapper = styled.h2`
  color: ${theme.darkColor};
  font-size: 1.8rem;
  font-weight: 550;
`;
const Location = styled.i`
  cursor: pointer;
  text-transform: capitalize;
`;
const Separator = styled.i`
  ::before {
    content: '/';
  }
  font-size: 2rem;
  margin: 0rem 0.2rem;
`;
