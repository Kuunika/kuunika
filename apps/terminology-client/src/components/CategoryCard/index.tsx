import React, { useState } from 'react';
import styled from 'styled-components';
import { theme } from '../../config/theme';
import Btn from '../Button';
import { ellipses } from '../../services/utils/helpers';
import { Collapse } from '@material-ui/core';
import { icon } from '@fortawesome/fontawesome-svg-core';

function CategoryCard({ title, content, onClick, icons }: Props) {
  const [more, setMore] = useState(false);
  const length = 100;
  const hasIcon = icons && icons.length > 0;
  const formatedContent = ellipses(content, length);
  return (
    <Wrapper data-testid="category-card" hasIcon={hasIcon}>
      <Title>{title}</Title>
      <ContentContainer>
        <Collapse in={more} collapsedHeight={130}>
          <Content>
            {more ? content : formatedContent}{' '}
            {content.length > length &&
              (more ? (
                <More onClick={() => setMore(false)}>Less</More>
              ) : (
                <More onClick={() => setMore(true)}>More</More>
              ))}
          </Content>
        </Collapse>
        <Icons count={icons.length}>
          {icons.map((icon, index) => (
            <i key={`${index}${icon.replace(/ /gi, '_')}`} className={icon}></i>
          ))}
        </Icons>
      </ContentContainer>
      <BtnContainer>
        <Btn onClick={onClick}>VIEW</Btn>
      </BtnContainer>
    </Wrapper>
  );
}

export default CategoryCard;

interface Props {
  title: string;
  content: any;
  icons?: Array<string>;
  onClick: Function;
}
const Wrapper = styled('div')<{ hasIcon: boolean }>`
  background: white;
  padding: 1.5rem;
  border-radius: 20px;
  box-shadow: 0px 3px 3px -2px rgba(0, 0, 0, 0.2),
    0px 3px 4px 0px rgba(0, 0, 0, 0.14), 0px 1px 8px 0px rgba(0, 0, 0, 0.12);
  text-align: ${props => (props.hasIcon ? 'left' : 'center')};
`;

const Title = styled.div`
  color: ${theme.darkColor};
  font-size: 2rem;
  font-weight: bold;
`;

const Content = styled.p`
  margin: 2rem 1rem 2rem 0rem;
  font-size: 1.2rem;
  font-weight: bold;
  color: gray;
  min-height: 92px;
  @media (max-width: 460px) {
    font-size: 1rem;
  }
`;

const More = styled.span`
  padding: 0.4rem;
  cursor: pointer;
  background: #ededed;
  font-size: 0.7rem;
  border-radius: 4px;
`;

const BtnContainer = styled.div`
  text-align: center;
  margin-top: 1rem;
`;

const ContentContainer = styled.div`
  display: flex;
  align-items: center;
  justify-content: center;
`;

const Icons = styled('div')<{ count: number }>`
  white-space: nowrap;
  i {
    font-size: ${props => (props.count > 1 ? '4.2rem' : '5rem')};
    color: #6e6c6a;
  }
`;
