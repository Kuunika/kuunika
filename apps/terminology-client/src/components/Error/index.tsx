import React from 'react';
import { State, IError } from '../../services/utils/@types';
import { useSelector } from 'react-redux';
import BaseError from './Base';
import {
  faNetworkWired,
  faExclamationTriangle
} from '@fortawesome/free-solid-svg-icons';
import { useHistory } from 'react-router-dom';

function Error() {
  const error: IError = useSelector((state: State) => state.errors.error);
  const history = useHistory();
  return (
    <>
      {error.status != 200 && (
        <BaseError
          message={error.message}
          icon={faExclamationTriangle}
          status={error.status}
          action={
            error.status == 404
              ? () => history.push('/')
              : () => location.reload()
          }
          actionText={error.status == 404 ? 'Go Home' : 'Refresh'}
        />
      )}
    </>
  );
}

export default Error;
