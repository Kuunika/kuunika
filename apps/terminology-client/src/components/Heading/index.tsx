import React, { useCallback, MouseEventHandler } from 'react';
import styled from 'styled-components';
import { useDispatch } from 'react-redux';
import { setActivePage } from '../../services/redux/actions/ui';
import { theme } from '../../config/theme';
import { Link } from 'react-router-dom';

function index() {
  const dispatch = useDispatch();
  const onClick = useCallback(
    item => {
      dispatch(setActivePage(''));
    },
    [dispatch]
  );
  return <SideBarView data-testid="side-bar" onLogoClick={onClick} />;
}

export default index;

function SideBarView({ onLogoClick }: { onLogoClick: Function }) {
  return (
    <Wrapper>
      <Link onClick={onLogoClick as MouseEventHandler<any>} to="/">
        <Logo src="../../assets/favicon.ico" />
      </Link>
      <SiteName>Malawi Terminology Dashboard</SiteName>
    </Wrapper>
  );
}

const Logo = styled.img`
  width: 120px;
  @media screen and (max-width: 960px) {
    width: 60px;
    display: inline-block;
  }
`;

const SiteName = styled.div`
  font-size: 30px;
  font-weight: bold;
  color: #231150;
  margin: 10px 0px;
  @media screen and (max-width: 960px) {
    font-size: 30px;
    display: inline-block;
    color: white;
    margin-left: 10px;
  }
  @media screen and (max-width: 500px) {
    font-size: 24px;
  }
  @media screen and (max-width: 460px) {
    font-size: 20px;
  }
`;

const Wrapper = styled.div`
  @media screen and (max-width: 960px) {
    display: flex;
    align-items: center;
    background: ${theme.color};
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    padding: 10px;
    z-index: 99;
    color: white;
  }
`;
