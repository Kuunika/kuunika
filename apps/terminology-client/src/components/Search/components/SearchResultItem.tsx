import React from 'react';
import styled from 'styled-components';
import Btn from '../../Button';
import { theme } from '../../../config/theme';
import { ISearchCategory } from 'apps/terminology-client/src/services/utils/@types';
import { useHistory } from 'react-router-dom';
import { useDispatch } from 'react-redux';
import { setSearchResultsState } from 'apps/terminology-client/src/services/redux/actions/ui';

function SearchResultItem({
  categoryBreadcrumb,
  numberOfResults,
  sourceId,
  searchTerm
}: ISearchCategory) {
  const history = useHistory();

  const dispatch = useDispatch();

  const onClick = () => {
    const link = `/${categoryBreadcrumb}/view/search/${sourceId}/${encodeURI(
      searchTerm
    )}`;
    history.push(link);
    dispatch(setSearchResultsState(false));
  };
  return (
    <Wrapper data-testid="search-result-item">
      <Category>
        Category: <i>{categoryBreadcrumb}</i>
        <Footer>
          <p>{numberOfResults} records found</p>
          <Btn
            style={{ width: '8rem', height: '2.5rem' }}
            onClick={onClick}
            data-testid="search-result-btn"
          >
            {' '}
            View{' '}
          </Btn>
        </Footer>
      </Category>
    </Wrapper>
  );
}

export default SearchResultItem;

const Wrapper = styled.div`
  padding: 1rem;
  display: flex;
  justify-content: space-between;
  align-items: center;
  background: white;
  border-radius: 10px;
  margin: 1rem;
`;

const Footer = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
  margin: 1rem 0rem;
`;

const Category = styled.div`
  text-align: left;
  width: 100%;
  margin: 0rem 1.5rem;
  font-size: 1.5rem;
  color: ${theme.darkColor};
  font-weight: 550;
  p {
    font-size: 1.2rem;
    color: gray;
  }
  @media (max-width: 460px) {
    font-size: 1rem;
    margin-right: 1rem;
    p {
      font-size: 0.8rem;
    }
  }
`;
