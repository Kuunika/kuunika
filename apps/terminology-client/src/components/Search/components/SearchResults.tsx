import React from 'react';
import styled from 'styled-components';
import SearchResultItem from './SearchResultItem';
import { useSelector } from 'react-redux';
import {
  State,
  ISearchCategory,
  ISearch
} from 'apps/terminology-client/src/services/utils/@types';
import { LinearProgress } from '@material-ui/core';

function SearchResults(props: Props) {
  const searchResults: ISearch = useSelector(
    (state: State) => state.data.search
  );

  const loading = useSelector((state: State) => state.loading.searchConcept);
  const error = useSelector((state: State) => state.errors.searchConcept || []);
  return (
    <>
      {props.open && searchResults.searchTerm.length > 0 && (
        <SearchResultsView
          data={searchResults}
          loading={loading}
          error={error}
        />
      )}
    </>
  );
}

export default SearchResults;

export function SearchResultsView({ data, loading, error }: ViewProps) {
  return (
    <Wrapper>
      {loading && (
        <div>
          <LinearProgress />
        </div>
      )}
      {data.searchCategories.length == 0 && !loading && (
        <div>No results for search</div>
      )}
      {error.length > 0 && <div>{error[0]}</div>}
      {data.searchCategories.map(result => (
        <SearchResultItem
          key={result.sourceId}
          {...result}
          searchTerm={data.searchTerm}
        />
      ))}
    </Wrapper>
  );
}

interface Props {
  open: boolean;
}

interface ViewProps {
  data: ISearch;
  loading: boolean;
  error: Array<string>;
}

const Wrapper = styled.div`
  padding: 1rem;
  min-width: 40rem;
  position: absolute;
  top: 110%;
  text-align: center;
  border-radius: 10px;
  z-index: 99;
  background: #f4f4f4;
  box-shadow: 0px 3px 3px -2px rgba(0, 0, 0, 0.2),
    0px 3px 4px 0px rgba(0, 0, 0, 0.14), 0px 1px 8px 0px rgba(0, 0, 0, 0.12);
`;
