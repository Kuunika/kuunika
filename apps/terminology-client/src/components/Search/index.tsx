import React, {
  useState,
  MouseEvent,
  MouseEventHandler,
  FocusEventHandler,
  useEffect,
  KeyboardEventHandler,
  useRef
} from 'react';
import styled from 'styled-components';
import { useSelector, useDispatch } from 'react-redux';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSearch, faTimes } from '@fortawesome/free-solid-svg-icons';
import ClickAwayListener from '@material-ui/core/ClickAwayListener';
import SearchResults from './components/SearchResults';
import {
  searchConcept,
  setSearchValue
} from '../../services/redux/actions/data';
import { setSearchResultsState } from '../../services/redux/actions/ui';
import { debounce } from 'lodash';
import { State } from '../../services/utils/@types';

function Search() {
  const open = useSelector((state: State) => state.ui.searchResults);

  const searchValue = useSelector(
    (state: State) => state.data.search.searchTerm
  );

  const dispatch = useDispatch();

  const handleClickAway = () => dispatch(setSearchResultsState(false));
  const onFocus = () => dispatch(setSearchResultsState(true));

  const onChange = (value = '') => {
    dispatch(setSearchValue(value));
    search(value);
  };
  const search = useRef(
    debounce(value => {
      dispatch(searchConcept(value));
    }, 800)
  ).current;

  return (
    <SearchView
      searchOpen={open}
      handleClickAway={handleClickAway}
      onFocus={onFocus}
      searchValue={searchValue}
      onChange={onChange}
    />
  );
}

export default Search;

export function SearchView(props: ViewProps) {
  return (
    <ClickAwayListener onClickAway={props.handleClickAway}>
      <Wrapper>
        <InputGroup>
          <Input
            placeholder="Search"
            onFocus={props.onFocus as FocusEventHandler<any>}
            onChange={e => props.onChange(e.target.value)}
            value={props.searchValue}
          ></Input>
          <Addon>
            {props.searchOpen && props.searchValue.length > 0 ? (
              <FontAwesomeIcon
                icon={faTimes}
                onClick={e => {
                  props.onChange(), props.handleClickAway(e as MouseEvent<any>);
                }}
              />
            ) : (
              <FontAwesomeIcon
                icon={faSearch}
                onClick={props.onFocus as MouseEventHandler<any>}
              />
            )}
          </Addon>
        </InputGroup>

        <SearchResults open={props.searchOpen} />
      </Wrapper>
    </ClickAwayListener>
  );
}

interface ViewProps {
  searchOpen: boolean;
  handleClickAway: (event: MouseEvent<Document>) => void;
  onFocus: Function;
  searchValue: string;
  onChange: Function;
}

const Wrapper = styled.div`
  text-align: center;
  display: flex;
  justify-content: center;
  position: relative;
  align-items: center;
  margin-bottom: 1rem;
`;

const InputGroup = styled.div`
  width: 35%;
  border-radius: 1.5rem;
  box-shadow: 0px 3px 3px -2px rgba(0, 0, 0, 0.2),
    0px 3px 4px 0px rgba(0, 0, 0, 0.14), 0px 1px 8px 0px rgba(0, 0, 0, 0.12);
  overflow: hidden;
  display: flex;
  background: white;
`;

const Input = styled.input`
  display: inline-block;
  background: white;
  border-width: 0px;
  height: 2.3rem;
  padding: 0.2rem;
  padding-left: 1.5rem;
  font-size: 1.5rem;
  width: 82.5%;
  border-radius: 1.5rem 0rem 0rem 1.5rem;
`;

const Addon = styled.div`
  display: inline-block;
  background: white;
  height: 2.3rem;
  display: flex;
  align-items: center;
  padding: 0.2rem;
  padding-right: 1.5rem;
  border-radius: 0rem 1.5rem 1.5rem 0rem;
  color: gray;
  cursor: pointer;
`;
