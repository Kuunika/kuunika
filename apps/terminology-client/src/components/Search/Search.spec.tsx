import React from 'react';
import { render } from '@testing-library/react';

import { SearchView } from './index';
import { SearchResultsView } from './components/SearchResults';
import Store from '../../services/redux/store';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { ISearch } from '../../services/utils/@types';

describe('Global Search', () => {
  it('should render search component', () => {
    const data = {
      searchOpen: false,
      handleClickAway: () => {},
      onFocus: () => {},
      searchValue: '',
      onChange: () => {}
    };
    const wrapper = render(
      <Provider store={Store}>
        <BrowserRouter>
          <SearchView {...data} />
        </BrowserRouter>
      </Provider>
    );

    expect(wrapper.baseElement).toMatchSnapshot();
  });
  it('should render search component open search results', () => {
    const data = {
      searchOpen: true,
      handleClickAway: () => {},
      onFocus: () => {},
      searchValue: 'hello',
      onChange: () => {}
    };
    const wrapper = render(
      <Provider store={Store}>
        <BrowserRouter>
          <SearchView {...data} />
        </BrowserRouter>
      </Provider>
    );

    expect(wrapper.baseElement).toMatchSnapshot();
  });
  it('should renders search results', () => {
    const data = {
      data: {
        searchTerm: 'Helloo',
        searchCategories: [
          {
            breadcrumbCategory: '/clinical',
            numberOfResults: 18,
            sourceId: '23233'
          },
          {
            breadcrumbCategory: '/clinicael',
            numberOfResults: 14,
            sourceId: '23233ds'
          }
        ]
      } as ISearch,
      loading: false,
      error: []
    };
    const wrapper = render(
      <Provider store={Store}>
        <BrowserRouter>
          <SearchResultsView {...data} />
        </BrowserRouter>
      </Provider>
    );

    expect(wrapper.baseElement).toMatchSnapshot();
  });
  it('should show loading state', () => {
    const data = {
      data: {
        searchTerm: 'Helloo',
        searchCategories: [
          {
            breadcrumbCategory: '/clinical',
            numberOfResults: 18,
            sourceId: '23233'
          },
          {
            breadcrumbCategory: '/clinicael',
            numberOfResults: 14,
            sourceId: '23233ds'
          }
        ]
      } as ISearch,
      loading: true,
      error: []
    };
    const wrapper = render(
      <Provider store={Store}>
        <BrowserRouter>
          <SearchResultsView {...data} />
        </BrowserRouter>
      </Provider>
    );

    expect(wrapper.baseElement).toMatchSnapshot();
  });
  it('should show no results on empty', () => {
    const data = {
      data: {
        searchTerm: 'Helloo',
        searchCategories: []
      } as ISearch,
      loading: false,
      error: []
    };
    const wrapper = render(
      <Provider store={Store}>
        <BrowserRouter>
          <SearchResultsView {...data} />
        </BrowserRouter>
      </Provider>
    );

    expect(wrapper.baseElement).toMatchSnapshot();
  });
  it('should show no results on error', () => {
    const data = {
      data: {
        searchTerm: 'Helloo',
        searchCategories: []
      } as ISearch,
      loading: false,
      error: ['Error']
    };
    const wrapper = render(
      <Provider store={Store}>
        <BrowserRouter>
          <SearchResultsView {...data} />
        </BrowserRouter>
      </Provider>
    );

    expect(wrapper.baseElement).toMatchSnapshot();
  });
});
