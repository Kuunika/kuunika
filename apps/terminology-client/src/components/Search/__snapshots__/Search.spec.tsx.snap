// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global Search should render search component 1`] = `
<body>
  <div>
    <div
      class="sc-ifAKCX eBsEho"
    >
      <div
        class="sc-EHOje bIBWcz"
      >
        <input
          class="sc-bZQynM iWWDmP"
          data-testid="search-box"
          placeholder="Search"
          value=""
        />
        <div
          class="sc-gzVnrw hUcRQU"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-search fa-w-16 "
            data-icon="search"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M505 442.7L405.3 343c-4.5-4.5-10.6-7-17-7H372c27.6-35.3 44-79.7 44-128C416 93.1 322.9 0 208 0S0 93.1 0 208s93.1 208 208 208c48.3 0 92.7-16.4 128-44v16.3c0 6.4 2.5 12.5 7 17l99.7 99.7c9.4 9.4 24.6 9.4 33.9 0l28.3-28.3c9.4-9.4 9.4-24.6.1-34zM208 336c-70.7 0-128-57.2-128-128 0-70.7 57.2-128 128-128 70.7 0 128 57.2 128 128 0 70.7-57.2 128-128 128z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Global Search should render search component open search results 1`] = `
<body>
  <div>
    <div
      class="sc-ifAKCX eBsEho"
    >
      <div
        class="sc-EHOje bIBWcz"
      >
        <input
          class="sc-bZQynM iWWDmP"
          data-testid="search-box"
          placeholder="Search"
          value="hello"
        />
        <div
          class="sc-gzVnrw hUcRQU"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-times fa-w-11 "
            data-icon="times"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 352 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M242.72 256l100.07-100.07c12.28-12.28 12.28-32.19 0-44.48l-22.24-22.24c-12.28-12.28-32.19-12.28-44.48 0L176 189.28 75.93 89.21c-12.28-12.28-32.19-12.28-44.48 0L9.21 111.45c-12.28 12.28-12.28 32.19 0 44.48L109.28 256 9.21 356.07c-12.28 12.28-12.28 32.19 0 44.48l22.24 22.24c12.28 12.28 32.2 12.28 44.48 0L176 322.72l100.07 100.07c12.28 12.28 32.2 12.28 44.48 0l22.24-22.24c12.28-12.28 12.28-32.19 0-44.48L242.72 256z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`Global Search should renders search results 1`] = `
<body>
  <div>
    <div
      class="sc-bxivhb gWqXUR"
      data-testid="search-container"
    >
      <div
        class="sc-bwzfXH eowHNW"
        data-testid="search-result-item"
      >
        <div
          class="sc-htpNat bNhkzF"
        >
          Category: 
          <i>
            /clinical
          </i>
          <p>
            18
             records found
          </p>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-1 MuiButton-contained"
          data-testid="search-result-btn"
          style="width: 8rem; height: 2.5rem;"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
             
            View
             
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="sc-bwzfXH eowHNW"
        data-testid="search-result-item"
      >
        <div
          class="sc-htpNat bNhkzF"
        >
          Category: 
          <i>
            /clinicael
          </i>
          <p>
            14
             records found
          </p>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-34 MuiButton-contained"
          data-testid="search-result-btn"
          style="width: 8rem; height: 2.5rem;"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
             
            View
             
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`Global Search should show loading state 1`] = `
<body>
  <div>
    <div
      class="sc-bxivhb gWqXUR"
      data-testid="search-container"
    >
      <div>
        <div
          class="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-indeterminate"
          role="progressbar"
        >
          <div
            class="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Indeterminate"
          />
          <div
            class="MuiLinearProgress-bar MuiLinearProgress-bar2Indeterminate MuiLinearProgress-barColorPrimary"
          />
        </div>
      </div>
      <div
        class="sc-bwzfXH eowHNW"
        data-testid="search-result-item"
      >
        <div
          class="sc-htpNat bNhkzF"
        >
          Category: 
          <i>
            /clinical
          </i>
          <p>
            18
             records found
          </p>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-66 MuiButton-contained"
          data-testid="search-result-btn"
          style="width: 8rem; height: 2.5rem;"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
             
            View
             
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
      <div
        class="sc-bwzfXH eowHNW"
        data-testid="search-result-item"
      >
        <div
          class="sc-htpNat bNhkzF"
        >
          Category: 
          <i>
            /clinicael
          </i>
          <p>
            14
             records found
          </p>
        </div>
        <button
          class="MuiButtonBase-root MuiButton-root WithStyles(ForwardRef(Button))-root-99 MuiButton-contained"
          data-testid="search-result-btn"
          style="width: 8rem; height: 2.5rem;"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiButton-label"
          >
             
            View
             
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</body>
`;

exports[`Global Search should show no results on empty 1`] = `
<body>
  <div>
    <div
      class="sc-bxivhb gWqXUR"
      data-testid="search-container"
    >
      <div>
        No results for search
      </div>
    </div>
  </div>
</body>
`;

exports[`Global Search should show no results on error 1`] = `
<body>
  <div>
    <div
      class="sc-bxivhb gWqXUR"
      data-testid="search-container"
    >
      <div>
        No results for search
      </div>
      <div>
        Error
      </div>
    </div>
  </div>
</body>
`;
